### PR TITLE
Add com.invoiceninja.InvoiceNinja

### PR DESCRIPTION
**Pull request by hillelcoren**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

The following permissions were added:

  - --filesystem=xdg-pictures
  - --filesystem=xdg-documents
  - --filesystem=xdg-download

The app supports uploading/downloading documents such as PDF invoices and expense receipt images.

- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.invoiceninja.app.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/main/data/flatpak-manifest.schema.json
+
+---
+app-id: com.invoiceninja.app
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
```
Comment (hfiguiere) [inline com.invoiceninja.app.yml:6]: current runtime is 23.08

File: com.invoiceninja.app.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
+        url: https://github.com/invoiceninja/admin-portal/releases/download/v5.0.128/Invoice-Ninja-Linux-Portable.tar.gz
+        sha256: 8275387f9f71a3c81df5f9932796dc23565fd7e4a56ab8705b1e214f86791511
+      - type: git
+        url: https://github.com/invoiceninja/admin-portal.git
+        branch: master
```
Comment (hfiguiere) [inline com.invoiceninja.app.yml:33]: NEVER a git branch. you need a tag or a commit.

File: com.invoiceninja.app.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - --filesystem=xdg-pictures
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-download
+modules:
+  - name: Invoice-Ninja
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
```
Comment (hfiguiere) [inline com.invoiceninja.app.yml:26]: why isn't this either upstream or a few commands inline.

File: com.invoiceninja.app.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    sources:
+      - type: file
+        url: https://github.com/invoiceninja/admin-portal/releases/download/v5.0.128/Invoice-Ninja-Linux-Portable.tar.gz
```
Comment (hfiguiere) [inline com.invoiceninja.app.yml:29]: good to see that portable mean x86_64. I thought it meant something else.

File: com.invoiceninja.app.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/main/data/flatpak-manifest.schema.json
+
+---
+app-id: com.invoiceninja.app
```
[the remark this answers is not in the document]
Comment (hillelcoren) [reply com.invoiceninja.app.yml:4]: I set the app-id to match the id we use on the Play Store

https://play.google.com/store/apps/details?id=com.invoiceninja.app

I can change to com.invoiceninja.InvoiceNinja if that's preferred?

File: com.invoiceninja.app.yml (hunk 1 above)
Comment (hillelcoren) [reply com.invoiceninja.app.yml:6]: Noted: I'll update it

File: com.invoiceninja.app.yml (hunk 3 above)
Comment (hillelcoren) [reply com.invoiceninja.app.yml:26]: I used this template, it's how they have it setup 

https://github.com/Merrit/flutter_flatpak_example

Comment (hfiguiere) [reply com.invoiceninja.app.yml:26]: two wrongs don't make a right.

File: com.invoiceninja.app.yml (hunk 2 above)
Comment (hillelcoren) [reply com.invoiceninja.app.yml:33]: Noted: I'll update it

File: com.invoiceninja.app.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (hillelcoren) [reply com.invoiceninja.app.yml:4]: Ok, got it

File: com.invoiceninja.app.yml (hunk 3 above)
Comment (hillelcoren) [reply com.invoiceninja.app.yml:26]: Does this need to be changed? I believe I saw other apps using the same approach.

Comment (hfiguiere) [reply com.invoiceninja.app.yml:26]: I believe I commented the same in the reviews and it got ignored.

File: com.invoiceninja.app.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (hillelcoren) [reply com.invoiceninja.app.yml]: This has been corrected

File: com.invoiceninja.app.yml (hunk 1 above)
Comment (hillelcoren) [reply com.invoiceninja.app.yml]: This has been corrected

File: com.invoiceninja.app.yml (hunk 2 above)
Comment (hillelcoren) [reply com.invoiceninja.app.yml]: This has been corrected

File: com.invoiceninja.app.yml (hunk 3 above)
Comment (hillelcoren) [reply com.invoiceninja.app.yml:26]: My preference is to leave it as is, if needs to be inlined to be accepted please let me know.

File: com.invoiceninja.app.yml
Comment (hfiguiere) [inline com.invoiceninja.app.yml:1]: the manifest filename must match the app-id.

Comment (hillelcoren) [reply com.invoiceninja.app.yml:1]: Thanks, fixed

**Comment by hfiguiere** [comment]
bot, build com.invoiceninja.InvoiceNinja

**Comment by hillelcoren** [comment]
Looks like it's failing here:

```
FB: Running 'flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/Invoice-Ninja --nofilesystem=host:reset --filesystem=/srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/build/Invoice-Ninja-1 --bind-mount=/run/build/Invoice-Ninja=/srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/build/Invoice-Ninja-1 --build-dir=/run/build/Invoice-Ninja --bind-mount=/run/ccache=/srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/ccache --unshare=network --env=SOURCE_DATE_EPOCH=1698856382 '--env=CFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer ' '--env=CXXFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer ' '--env=LDFLAGS=-L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed ' --env=CCACHE_DIR=/run/ccache/disabled --env=PATH=/app/bin:/usr/bin --env=LD_LIBRARY_PATH=/app/lib --env=PKG_CONFIG_PATH=/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig --env=FLATPAK_BUILDER_N_JOBS=64 /srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/rofiles/rofiles-fnbZe7 /bin/sh -c ./build-flatpak.sh' on host
/bin/sh: line 1: ./build-flatpak.sh: Permission denied
```

Do you have any suggestions?

**Comment by hillelcoren** [comment]
I think I see the issue, I'll check in a fix

**Comment by hillelcoren** [comment]
bot, build com.invoiceninja.InvoiceNinja

**Comment by hillelcoren** [comment]
bot, build com.invoiceninja.InvoiceNinja

**Comment by hillelcoren** [comment]
bot, build com.invoiceninja.InvoiceNinja

**Comment by hillelcoren** [comment]
bot, build com.invoiceninja.InvoiceNinja

**Comment by hillelcoren** [comment]
bot, build com.invoiceninja.InvoiceNinja

**Comment by hfiguiere** [comment]
The list of valid SPDX license identifier is at
https://spdx.org/licenses/

**Comment by hillelcoren** [comment]
Thanks! The app license is AAL which I see listed here however it failed with this error:

• tag-invalid           : <metadata_license> is not valid [AAL]

Are both metadata_license and project_license required?

**Comment by hfiguiere** [comment]
Some licenses are not acceptable for the metadata. `CC0-1.0` is preferred.

**Comment by hfiguiere** [comment]
this also implies you have the right place it under this license. ie if you are not the rights-holder you can't unilaterally do it.

**Comment by hillelcoren** [comment]
bot, build com.invoiceninja.InvoiceNinja

**Comment by hillelcoren** [comment]
Thanks for your help! I've tested the app, it looks good. 

If you'd like to try it here are demo credentials:

```
Email: demo@invoiceninja.com
Password: Password0
URL: demo.invoiceninja.com
```

What's the next step?

**Pull request closed without merging**